File: chromeos/components/camera_app_ui/resources/js/externs/chrome.js
// Copyright 2019 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

/**
 * @typedef {{
 *   get: function(Array<string>, function(Object)),
 *   set: function(string, ?),
 * }}
 */
chrome.chromeosInfoPrivate;

/** @const */
chrome.events = {};

/**
 * @typedef {{
 *   addListener: function(function(): void),
 *   removeListener: function(function(): void),
 * }}
 */
chrome.events.Event;

File: tools/cr/cr/targets/chrome.py
# Copyright 2013 The Chromium Authors
# Use of this source code is governed by a BSD-style license that can be
# found in the LICENSE file.

"""Module for the chrome targets."""

import cr


class ChromeTarget(cr.NamedTarget):
  NAME = 'chrome'
  CONFIG = cr.Config.From(
      CR_RUN_ARGUMENTS=cr.Config.Optional('-d "{CR_URL!e}"'),
      CR_TARGET_NAME='Chrome',
  )


class ChromeTestTarget(cr.NamedTarget):
  NAME = 'chrome_test'
  CONFIG = cr.Config.From(
      CR_TARGET_NAME='ChromeTest',
      CR_TEST_TYPE=cr.Target.INSTRUMENTATION_TEST,
      CR_RUN_DEPENDENCIES=[ChromeTarget.NAME],
  )
